### ISSUE TITLE
MD053 / Horizontal Rule style complains when horizontal rule in blockquotes

### ISSUE THREAD

**Issue by redactedscribe** [DavidAnson/markdownlint#430]
I'd like to use a horizontal rule inside of blockquotes to help with separation without having to resort to needing to place some text between two separate blockquotes (as suggested in [MD028](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md028---blank-line-inside-blockquote)).

```markdown
L1  > Some text.
L2  >
L3  > ---
L4  > Some more text.
```

Problem is, line 3 complains about consistency due to a plain `---` already existing outside of the block quote.

I've tried:

```json
"markdownlint.config": {
    "hr-style": {
        "style": "> ---"
    }
}
```

But this of course only makes the plain `---` to be the one to complain.

Is there some way to specify multiple valid `style` values? Or maybe the MD rule can be improved to account for this use case (I don't know its implications for all the markdown interpreters)? The "consistent" value should mean the use of the same style of horizontal rule, `---` vs `***`, rather than if it's inside a blockquote or not, right?

I'm not sure if it's expected, but, if line 2 isn't included as a spacer, line 1 complains with:
```
MD003/heading-style/header-style: Heading style [Expected: atx; Actual: setext]
```
and
```
MD022/blanks-around-headings/blanks-around-headers: Headings should be surrounded by blank lines`.
```
Is `>` a type of header, or is this a bug? I can live with inserting a blank line, but I thought I'd mention this just in case.

Thanks.

markdownlint v0.44.0

**Comment by DavidAnson** [comment]
The block quoted HR should not be reported as a consistency violation. Here's a reproduction: https://dlaa.me/markdownlint/#%25m%23%20Issue%20430%0A%0A---%0A%0A%3E%20Text%0A%3E%0A%3E%20---%0A%3E%20Text%0A

**Comment by DavidAnson** [comment]
The messages when blank line L2 are not present are correct; that syntax is interpreted as a setext heading (note bold L1 text): https://dlaa.me/markdownlint/#%25m%23%20Issue%20430%0A%0A%3E%20Text%0A%3E%20---%0A%3E%20Text%0A

**Comment by redactedscribe** [comment]
Hello @DavidAnson, thanks for the fix.

I was wondering if this will be applied to vscode-markdownlint as well? At the time of posting, I wasn't aware that there was also a vscode repo.

**Comment by DavidAnson** [comment]
Yes. Next update will be to markdownlint-cli2, then the  VS Code extension.

**Comment by redactedscribe** [comment]
Great news! And by the looks of it, you're closing in on being able to fix the presence of parentheses in paths issue. Looking forward to being able to make use of this extension again (most my *.md files reside within such a path).
